apps/settings/js/panels/operator_settings/models/operator_manager.js

```diff
@@ -0,0 +1,205 @@
+   * @type {OperatorManager.AUTO_SELECTION_STATE}
+   */
+  Observable.defineObservableProperty(OperatorManager.prototype,
+    'autoSelectionState', {
+      readonly: true,
+      value: null
+  });
+
+  /**
+   * An observable property indicating the currently connected operator.
+   *
+   * @access public
+   * @readonly
+   * @memberOf OperatorManager.prototype
+   * @type {OperatorItem}
+   */
+  Observable.defineObservableProperty(OperatorManager.prototype,
+    'connectedOperatorItem', {
+      readonly: true,
+      value: null
+  });
+
+  /**
+   * An observable property indicating the searching status.
+   *
+   * @access public
+   * @readonly
+   * @memberOf OperatorManager.prototype
+   * @type {Boolean}
+   */
+  Observable.defineObservableProperty(OperatorManager.prototype,
+    'searching', {
+      readonly: true,
+      value: false
+  });
+
+  /**
+   * Request to search available operators and update the operator list.
+   *
+   * @access public
+   * @memberOf OperatorManager.prototype
+   */
+  OperatorManager.prototype.search = function() {
+    this._operators.reset([]);
+    this._searching = true;
+    this._connWrapper.search().then((networks) => {
+      this._searching = false;
+      networks = networks || [];
+      var operators = networks.map((network) => OperatorItem(network));
+      operators.forEach((operator) => {
+        if (operator.network.state === OperatorItem.STATE.CONNECTED) {
+          this._connectedOperatorItem = operator;
+        }
+      });
+      this._operators.reset(operators);
+    }).catch((error) => {
+      this._searching = false;
+      this.error('could not retrieve any network operator: ' + error);
+    });
+  };
+
+  /**
+   * Stop the search.
+   *
+   * @access public
+   * @memberOf OperatorManager.prototype
+   */
+  OperatorManager.prototype.stop = function() {
+    this._connWrapper.stop();
+    this._searching = false;
+  };
+
+  /**
+   * Connect to a network. If it fails, reconnect to the original operator 5
+   * seconds later.
+   *
+   * @access public
+   * @memberOf OperatorManager.prototype
+   * @params {OperatorItem} operatorItem
+   */
+  OperatorManager.prototype.connect = function(operatorItem) {
+    this.debug('connect');
+    operatorItem.setState(OperatorItem.STATE.CONNECTING);
+    this._connWrapper.connect(operatorItem.network).then(() => {
+      operatorItem.setState(OperatorItem.STATE.CONNECTED);
+      this._connectedOperatorItem = operatorItem;
+    }).catch((error) => {
+      this.error('unable to connect to the network: ' + error);
+      operatorItem.setState(OperatorItem.STATE.FAILED);
+      // recover the connection
+      setTimeout(function(originalOperatorItem) {
+        if (this.autoSelectionState === AutoSelectionModel.STATE.DISABLED) {
+          operatorItem.setState(OperatorItem.STATE.IDLE);
+          this.connect(originalOperatorItem);
+        }
+      }.bind(this, this._connectedOperatorItem), 5000);
```

Comment:
is it possible that users have no simcards and has no way to have this pre-connected operator item ? If we do have this case, maybe we have to check existence of  `originalOperatorItem` before calling `this.connect` ?